src/Compilers/CSharp/Portable/Binder/UsingStatementBinder.cs

```diff
@@ -132,9 +131,33 @@ internal override BoundStatement BindUsingStatementParts(DiagnosticBag diagnosti
                 expressionOpt,
                 iDisposableConversion,
                 boundBody,
+                disposeMethod, // This ensures that a pattern-matched Dispose statement is not used.
                 hasErrors);
         }
 
+        /// <summary>
+        /// Checks for a Dispose method on exprType. Failing to satisfy the pattern is not an error -
+        /// it just means we have to check for an interface instead.
+        /// </summary>
+        /// <param name="exprType">Type of the expression over which to iterate</param>
+        /// <param name="diagnostics">Populated with warnings if there are near misses</param>
+        /// <returns>True if a matching method is found (still need to verify return type).</returns>
+        private MethodSymbol TryFindDisposePatternMethod(TypeSymbol exprType, DiagnosticBag diagnostics)
+        {
+            LookupResult lookupResult = LookupResult.GetInstance();
+            SyntaxNode exp = _syntax.Expression != null ? (SyntaxNode) _syntax.Expression : (SyntaxNode) _syntax.Declaration;
+            MethodSymbol disposeMethod = FindPatternMethod(exprType, WellKnownMemberNames.DisposeMethodName, lookupResult, exp, warningsOnly: true, diagnostics: diagnostics, _syntax.SyntaxTree);
```

Comment:
>FindPatternMethod [](start = 41, length = 17)

Are there cases where we benefit from having the `LookupResult` allocated in the caller? If not it might be better to push it into the method.